rules_building_block/discovery_userdata_request_from_ec2_instance.toml

```diff
@@ -0,0 +1,61 @@
+[metadata]
+creation_date = "2024/04/14"
+integration = ["aws"]
+maturity = "production"
+min_stack_comments = "New fields added: required_fields, related_integrations, setup"
+min_stack_version = "8.9.0"
+updated_date = "2024/04/14"
+
+[rule]
+author = ["Elastic"]
+building_block_type = "default"
+description = """
+Identifies discovery request `DescribeInstanceAttribute` with the attribute userData and instanceId in AWS CloudTrail
+logs. This may indicate an attempt to retrieve user data from an EC2 instance. Adversaries may use this information to
+gather sensitive data from the instance or to identify potential vulnerabilities. This is a building block rule that
+does not generate an alert on its own, but serves as a signal for anomalous activity.
```

Comment:
If this is the case, is this a higher-order rule? Or are higher-order rules, the rules that query this rule's output?

Reply:
Good question! I'd say this is a good signal for either threat hunting, TDIR or high-order rules but definitely not a high confident indicator of compromise.